### PR TITLE
redo (ticdc): only delete changefeed related file when changefeed was removed

### LINKED ISSUE #6413: redo manager should only clean files of one specified changefeed instead of all changefeeds
### Before asking a question, make sure you have

- [X] Searched existing Stack Overflow questions
- [X] Googled your question
- [X] Searched open and closed [GitHub issues](https://github.com/pingcap/tiflow/issues?q=is%3Aissue)
- [X] Read the [documentation](https://docs.pingcap.com/tidb/stable)

### What is your question?

Test shows there is a bug about cleaning redo-log files when removing a changefeed:

* when removing a changefeed, `managerImpl.Cleanup` will be called;
* then `LogWriter.DeleteAllLogs` will be called, and all files in the redo-log directory and S3 path will be deleted;
* so if **several changefeeds share one redo-log directory or S3 bucket**, their redo files will be deleted incorrectly.

The bug should effect all versions.

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #6413 

### What is changed and how it works?

Only delete the file belong to a changefeed when it's release, rather than the whole directory that used by it.

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
1. create changefeed test-1, and other two changefeed. Enable redo log for all of them, using the same storage uri:
```
s3://guagua/test1?access-key=xxxx&secret-access-key=xxxx&endpoint=http://xxxx/
```
2. Replicate some data, and remove one changefeed.
3. Inspect the `s3://guagua/test1` dir, the files of removed changefeed has been deleted but other two changefeeds' files remain.
image

### Release note 

```release-note
Fix a bug that may cause changefeed's redo log files be deleted wrongly.
```

### CODE REVIEW

File: cdc/redo/common/util.go [hunk 1]
```diff
@@ -160,3 +162,23 @@ func defaultS3Retryer() request.Retryer {
 		},
 	}
 }
+
+// GetChangefeedFiles return the files that belong to the changefeed.
+func GetChangefeedFiles(files []string, changefeedID model.ChangeFeedID) []string {
```
Comment (nongfushanquan) [inline cdc/redo/common/util.go:167]: Maybe the name such as FilterChangefeedFiles  is more proper for this method

**Review by nongfushanquan — COMMENTED** [review]
please correct the title "Redo"

File: cdc/redo/writer/file.go [hunk 2]
```diff
@@ -446,6 +446,7 @@ func (w *Writer) newPageWriter() error {
 	return nil
 }
 
+// why we call it rotate?
```
Comment (zhaoxinyu) [inline cdc/redo/writer/file.go:449]: Because this function will close the current file and open a new file.

File: cdc/redo/writer/writer.go [hunk 3]
```diff
@@ -51,7 +51,7 @@ type RedoLogWriter interface {
 	// Regressions on them will be ignored.
 	FlushLog(ctx context.Context, checkpointTs, resolvedTs model.Ts) error
 
-	// Get the current meta.
+	// GetMeta get current meta.
```
Comment (zhaoxinyu) [inline cdc/redo/writer/writer.go:54]:
```suggestion
	// GetMeta gets current meta.
```

File: cdc/redo/writer/writer.go [hunk 4]
```diff
@@ -93,7 +93,9 @@ type LogWriter struct {
 	cfg       *LogWriterConfig
 	rowWriter fileWriter
 	ddlWriter fileWriter
-	storage   storage.ExternalStorage
+	// storage in LogWriter is used to clean up the
```
Comment (zhaoxinyu) [inline cdc/redo/writer/writer.go:96]: Maybe this comment is not accurate. Because use also use `storage` to save metadata. See function `writeMetaToS3()`

**Comment by zhaoxinyu** [comment]
Rest LGTM

File: pkg/uuid/uuid.go [hunk 5]
```diff
@@ -15,7 +15,7 @@ package uuid
 
 import guuid "github.com/google/uuid"
 
-// Generator defines an interface that can generate a uuid
+// Generator defines an interface that can generate an uuid
```
Comment (zhaoxinyu) [inline pkg/uuid/uuid.go:18]:
```suggestion
// Generator defines an interface that can generate a uuid
```

Comment (zhaoxinyu) [reply pkg/uuid/uuid.go:18]: The original code comment is correct.

**Review by zhaoxinyu — APPROVED** [review]

File: cdc/redo/common/util.go [hunk 6]
```diff
@@ -21,6 +21,8 @@ import (
 	"strings"
 	"time"
 
+	"github.com/pingcap/tiflow/cdc/model"
+
```
Comment (hicqu) [inline cdc/redo/common/util.go:25]: How about removing the empty line?

File: cdc/redo/writer/writer.go [hunk 7]
```diff
@@ -340,27 +342,50 @@ func (l *LogWriter) DeleteAllLogs(ctx context.Context) (err error) {
...
+	}
+	filteredFiles := common.FilterChangefeedFiles(fileNames, l.cfg.ChangeFeedID)
+
+	if len(filteredFiles) == len(fileNames) {
+		if err = os.RemoveAll(l.cfg.Dir); err != nil {
+			return cerror.WrapError(cerror.ErrRedoFileOp, err)
+		}
+	} else {
+		for _, file := range filteredFiles {
+			if err = os.RemoveAll(filepath.Join(l.cfg.Dir, file)); err != nil {
```
Comment (hicqu) [inline cdc/redo/writer/writer.go:367]: If the error is `NotExists`, it should be ignored.

File: cdc/redo/common/util.go (hunk 6 above)
Comment (asddongmen) [reply cdc/redo/common/util.go:25]: good catch

**Review by nongfushanquan — APPROVED** [review]

**Review by hicqu — APPROVED** [review]

**Pull request merged**